Mathlib/CategoryTheory/Functor/KanExtension/Preserves.lean

```diff
@@ -0,0 +1,281 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Limits.Preserves.Basic
+
+/-!
+# Preservation of Kan extensions
+
+Given functors `F : A ⥤ B`, `L : B ⥤ C`, and `G : B ⥤ D`,
+we introduce a typeclass `G.PreservesLeftKanExtension F L`, which encodes the fact that
+the left/right Kan extension of `F` along `L` is preserved by the functor `G`.
+
+When the Kan extension is pointwise, It suffices that `G` preserves (co)limits of the relevant
+diagrams.
+
+-/
+
+namespace CategoryTheory.Functor
+
+variable {A B C D : Type*} [Category A] [Category B] [Category C] [Category D]
+  (G : B ⥤ D) (F : A ⥤ B) (L : A ⥤ C)
+
+section LeftKanExtension
+
+/-- `G.PreservesLeftKanExtension F L` asserts that `G` preserves all left kan extensions
+of `F` along `L`. -/
+class PreservesLeftKanExtension where
+  preserves : ∀ (F' : C ⥤ B) (α : F ⟶ L ⋙ F') [IsLeftKanExtension F' α],
+    IsLeftKanExtension (F' ⋙ G) <| whiskerRight α G ≫ (Functor.associator _ _ _).hom
```

Comment:
When I thought about preservation of Kan extensions many months ago I chose the definition
```lean
class PreservesLeftKanExtension (L : C ⥤ D) (F : C ⥤ H) (G : H ⥤ I) : Prop where
  preserves : ∀ {E : LeftExtension L F},
    E.IsUniversal → Nonempty ((LeftExtension.postcomp₂ G).obj E).IsUniversal
```
(my `postcomp₂` is your `whiskerRight`). Both definitions seem reasonable to me at first glance. Should the `IsUniversal` phrasing be provided as an alternative constructor/consequence? I haven't used the Kan extension library too much, so I'm not sure if this would be useful to have.